### PR TITLE
Azure: Improve handling of deprecated query types

### PR DESCRIPTION
**Pull request by aangelisc**
Two query types have been deprecated for awhile in Azure: `Application Insights` and `Insights Analytics`. If a user attempts to submit a query with either of these values set as the type then the error is incorrectly processed as a plugin error when it should be a downstream error as migration information is available [here](https://grafana.com/docs/grafana/v9.0/datasources/azuremonitor/deprecated-application-insights/).

This change improves the logic and will appropriately mark these errors as downstream whilst still informing the user of the error.

### CODE REVIEW

**Review by wbrowne — APPROVED** [review]
LGTM (as long as you want to nuke the entire request which may have some valid/non-deprecated queries)

File: pkg/tsdb/azuremonitor/azuremonitor.go [hunk 1]
```diff
@@ -54,7 +54,30 @@ func ProvideService(httpClientProvider *httpclient.Provider) *Service {
 	return s
 }
 
+func handleDeprecatedQueryTypes(req *backend.QueryDataRequest) *backend.QueryDataResponse {
+	// Logic to handle deprecated query types that haven't been migrated
+	responses := backend.Responses{}
+	for _, q := range req.Queries {
+		if q.QueryType == "Application Insights" || q.QueryType == "Insights Analytics" {
+			responses[q.RefID] = backend.DataResponse{
+				Error:       fmt.Errorf("query type: '%s' is no longer supported. Please migrate this query", q.QueryType),
```
Comment (bossinc) [inline pkg/tsdb/azuremonitor/azuremonitor.go:63]: Should we give them more info on migrating the query? Are the ways to migrate too varied?

**Review by bossinc — APPROVED** [review]

**Comment by bossinc** [comment]
I like the new tests!

**Comment by aangelisc** [comment]
> LGTM (as long as you want to nuke the entire request which may have some valid/non-deprecated queries)

I think it's okay to nuke the request as it'll encourage the user to migrate 😊

File: pkg/tsdb/azuremonitor/azuremonitor.go (hunk 1 above)
Comment (aangelisc) [reply pkg/tsdb/azuremonitor/azuremonitor.go]: I've added a link to the docs detailing the migration info 😊

**Pull request merged**